### PR TITLE
Setup codecov

### LINKED ISSUE #1441: Set up codecov

### LINKED ISSUE #1834: React warning
I'm seeing this warning and a bunch of similar ones after #1787 was merged, under LearnerSearchPage:

Warning: You are manually calling a React.PropTypes validation function for the `children` prop on `Slider`. This is deprecated and will not work in production with the next major version. You may be seeing this warning due to a third-party PropTypes library. See https://fb.me/react-warning-dont-call-proptypes for details.

### PR DESCRIPTION
**Pull request by noisecapella**
#### What are the relevant tickets?
Fixes #1441 
Fixes #1834

#### What's this PR do?
Changes settings to upload results to codecov.

#### How should this be manually tested?
This shouldn't touch any non-test code so it shouldn't need manual testing.

To look at the coverage results, go to the root for this github repository, switch to branch `gs/codecov6` and click on the coverage button. You should see both javascript and python files in the files listed.

#### Any background context you want to provide?
- The `.dockerignore` change is necessary because we need to copy `.git` so `codecov` can make use of it. 

#### What GIF best describes this PR or how it makes you feel?
https://twitter.com/JoachimHolmer/status/799602767081848832/photo/1

### CODE REVIEW

File: static/js/components/search/FilterVisibilityToggle_test.js [hunk 1]
```diff
@@ -3,19 +3,32 @@ import React from 'react';
 import { assert } from 'chai';
 import { mount } from 'enzyme';
 import sinon from 'sinon';
+import { SearchkitManager, SearchkitProvider } from 'searchkit';
```
Comment (aliceriot) [inline static/js/components/search/FilterVisibilityToggle_test.js]: does this fix that warning? If so :100:

Comment (noisecapella) [inline static/js/components/search/FilterVisibilityToggle_test.js]: Yes that's the intention here

**Comment by noisecapella** [comment]
yay code coverage

**Comment by rhysyngsun** [comment]
:+1:

**Pull request merged**